### PR TITLE
ROC - 5042 claimant dashboard update for company part admit rejection

### PR DESCRIPTION
**Pull request by rattur**
### JIRA link

https://tools.hmcts.net/jira/browse/ROC-5042

### Change description

Update claimant dashboard and claim status when a claimant rejects a part admission company defendant

### Work checklist

- [ ] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [ x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x ] No

### CODE REVIEW

**Comment by singh-surjeet** [comment]
@rattur have you attempted putting in part admit details block in file partAdmitRejectionDefendant.njk
{% if claim.claimantResponse.freeMediation === FreeMediationOption.YES %}
Problem with nunjucks is - It's not good with enums. Look at this PR - https://github.com/hmcts/cmc-citizen-frontend/pull/1329/files

File: src/main/app/claims/models/claimStatus.ts [hunk 1]
```diff
@@ -20,6 +20,8 @@ export enum ClaimStatus {
   CLAIMANT_ACCEPTED_ADMISSION_AND_REQUESTED_CCJ,
   ADMISSION_SETTLEMENT_AGREEMENT_REACHED,
   CLAIMANT_REJECTED_DEFENDANT_AS_BUSINESS_RESPONSE,
+  CLAIMANT_REJECTED_DEFENDANT_AS_BUSINESS_RESPONSE_WITH_MEDIATION,
```
Comment (singh-surjeet) [inline src/main/app/claims/models/claimStatus.ts]: I question we require new statuses. Please look at comments.

Comment (rattur) [reply src/main/app/claims/models/claimStatus.ts]: Tested with FreeMediationOption.YES and everything works fine, however, if I remove the additional statuses the correct content doesn't show, therefore I think we need the extra statuses

Comment (robertparkinson) [reply src/main/app/claims/models/claimStatus.ts]: I agree with surjeet. These statuses are using the same content as other statuses and shouldn't be needed. You should just need to update hasClaimantRejectedPartAdmission if the status is not showing up

Comment (rattur) [reply src/main/app/claims/models/claimStatus.ts]: Ok let me have a look

File: src/main/features/dashboard/views/macro/claimStatus/partAdmitRejectionClaimant.njk [hunk 2]
```diff
@@ -22,5 +22,11 @@
     {% endif %}
 
     <p>{{ downloadResponseLink(claim) }}</p>
+
+    <div class="status-block">
```
[the remark this answers is not in the document]
Comment (rattur) [reply src/main/features/dashboard/views/macro/claimStatus/partAdmitRejectionClaimant.njk]: Removed

File: src/main/features/dashboard/views/macro/caseHistory.njk [hunk 3]
```diff
@@ -21,6 +21,7 @@ ccjRequestedForDefendantClaimDetails %}
 {% from "./claimStatus/claimPartAdmittedDefendant.njk" import
   claimPartAdmittedToOweForDefendantClaimDetails,
   claimPartAdmittedAlreadyPaidForDefendantClaimDetails,
+  claimPartAdmittedToOweForDefendantDashboard,
```
Comment (robertparkinson) [inline src/main/features/dashboard/views/macro/caseHistory.njk]: this is never used, please remove.

File: src/main/features/dashboard/views/macro/caseStatus.njk [hunk 4]
```diff
@@ -134,6 +134,8 @@
     {{ claimantWithDefendantAsBusinessResponse() }}
     {% elif status === ClaimStatus.CLAIMANT_ACCEPTED_DEFENDANT_PART_ADMISSION_AS_BUSINESS_WITH_ALTERNATIVE_PAYMENT_INTENTION_RESPONSE %}
     {{ claimantWithDefendantAsBusinessResponse() }}
+    {% elif ClaimStatus.CLAIMANT_REJECTED_DEFENDANT_AS_BUSINESS_RESPONSE_WITH_MEDIATION %}
```
Comment (robertparkinson) [inline src/main/features/dashboard/views/macro/caseStatus.njk]: this status doesn't exist any more please remove this if as the existing part admission code should now take care of it

Comment (rattur) [reply src/main/features/dashboard/views/macro/caseStatus.njk]: done

File: src/main/features/dashboard/views/macro/caseHistory.njk (hunk 3 above)
Comment (rattur) [reply src/main/features/dashboard/views/macro/caseHistory.njk]: done

**Pull request merged**